### PR TITLE
Answer like menu added. Fixes #843

### LINKED ISSUE #843: List of who pressed "like" should be shown on each answer in Q&A
This has been marked as a good candidate for becoming a [first-timers-only issue](http://www.firsttimersonly.com) like [these](https://github.com/publiclab/plots2/labels/first-timers-only), meaning that it's simple, self-contained, and with some extra formatting, could be a great entry point for a new contributor. If you're familiar enough with this code, please consider [reformatting or reposting it as a first-timers-only issue](https://publiclab.org/notes/warren/10-31-2016/create-a-welcoming-first-timers-only-issue-to-invite-new-software-contributors), and then ping @jywarren to get it labelled.

****
Our new [Q&A system](https://publiclab.org/questions) is great, and people can "like" individual answers -- like one of the answers on this page:

https://publiclab.org/questions/gretchengehrke/09-20-2016/question-would-it-be-feasible-to-use-a-stationary-camera-to-conduct-emissions-opacity-evaluation

screenshot 2016-09-26 at 4 33 30 pm

However, you can't tell who else liked the answer -- the way you can see who liked a question (or note):

screenshot 2016-09-26 at 4 34 16 pm

This might take some UI design work -- maybe a caret menu like how it's implemented for questions:

https://github.com/publiclab/plots2/blob/master/app/views/like/_like.html.erb#L8-L14

### PR DESCRIPTION
**Pull request by Pk13055**
#843 
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] all tests pass -- `rake test:all`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request is descriptively named with #number reference back to original issue
* [x] if possible, multiple commits squashed if they're smaller changes
* [x] reviewed/confirmed/tested by another contributor or maintainer
* [x] `schema.rb.example` has been updated if any database migrations were added

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

**Comment by Pk13055** [comment]
@ryzokuken Can you review the change?

**Comment by ryzokuken** [comment]
@Pk13055 LGTM, why did you add the newline though. Seems unnecessary.

File: app/views/questions/_answer.html.erb [hunk 1]
```diff
@@ -43,8 +43,17 @@
...
+        <div class="btn btn-default btn-sm" rel="popover" data-placement="left" data-html="true" data-title="Users who like this" data-content="
+            <% answer.likers.each do |user| %>
+                <i class='fa fa-star-o'></i> <a href='/profile/<%= user.username %>/likes'><%= user.username %></a></br>
+            <% end %>
+        "><span class="caret"></span>
+        </div>
+    <% end %>
        <a href=<% if current_user %>"/answer_like/likes/<%= answer.id %>"<% else %>"#"<% end %>data-remote ="true" tooltip-title="Helpful? Like it" class="btn btn-default btn-sm btn-like" id="answer-like-button-<%= answer.id %>" <% unless current_user %> popover-content="You must be logged in to Like the answer" <% end %> ><span id="answer-like-star-<%= answer.id %>" class="fa fa-star<% if current_user && !answer.liked_by(current_user.uid) %>-o<% end %>"></span> <span id="answer-like-count-<%= answer.id %>"><%= answer.likers.length %></span> Likes</a>
-      <a href="#a<%= answer.id %>" class="btn btn-default btn-sm "><i class="fa fa-share-alt"></i></a> &nbsp;
+
```
Comment (ryzokuken) [inline app/views/questions/_answer.html.erb]: Why did you add this newline? Seems unnecessary.

File: app/views/questions/_answer.html.erb [hunk 2]
```diff
@@ -43,8 +43,17 @@
           <a href="/answers/accept/<%= answer.id %>" data-remote="true" id="answer-<%= answer.id %>-accept" class="answer-accept btn btn-sm btn-default">Accept this answer</a>  
         <% end %>
       <% end %>
+    <% if answer.likers.length != 0 %>
+        <div class="btn btn-default btn-sm" rel="popover" data-placement="left" data-html="true" data-title="Users who like this" data-content="
+            <% answer.likers.each do |user| %>
```
Comment (ryzokuken) [inline app/views/questions/_answer.html.erb]: I might be overstepping myself over here, but what if a lot of users like the same post? We shouldn't really name each of the 100+ people who liked it but perhaps we could always show the first 5-10 people and a `+100`-like link which triggers a modal which shows the entirety of people who liked it. 

@jywarren What do you say?

Comment (Pk13055) [inline app/views/questions/_answer.html.erb]: @jywarren If we do go ahead with the modal, I think it makes sense to incorporate it into the 'notes' liked menu as well.
@ryzokuken How about opening up another issue?

File: app/views/questions/_answer.html.erb [hunk 3]
```diff
@@ -43,8 +43,16 @@
...
+                <i class='fa fa-star-o'></i> <a href='/profile/<%= user.username %>/likes'><%= user.username %></a></br>
+            <% end %>
+        "><span class="caret"></span>
+        </div>
+    <% end %>
        <a href=<% if current_user %>"/answer_like/likes/<%= answer.id %>"<% else %>"#"<% end %>data-remote ="true" tooltip-title="Helpful? Like it" class="btn btn-default btn-sm btn-like" id="answer-like-button-<%= answer.id %>" <% unless current_user %> popover-content="You must be logged in to Like the answer" <% end %> ><span id="answer-like-star-<%= answer.id %>" class="fa fa-star<% if current_user && !answer.liked_by(current_user.uid) %>-o<% end %>"></span> <span id="answer-like-count-<%= answer.id %>"><%= answer.likers.length %></span> Likes</a>
-      <a href="#a<%= answer.id %>" class="btn btn-default btn-sm "><i class="fa fa-share-alt"></i></a> &nbsp;
+        <a href="#a<%= answer.id %>" class="btn btn-default btn-sm "><i class="fa fa-share-alt"></i></a> &nbsp;
```
Comment (ryzokuken) [inline app/views/questions/_answer.html.erb]: @Pk13055 Sorry to bother you again, but unnecessary space here :P

**Comment by jywarren** [comment]
I think we haven't had an instance of too many likes yet, but couldn't we
make the pop-up scrollable? But also I agree it should be a separate issue
and we should merge this one. Thanks!

**Comment by Pk13055** [comment]
@jywarren Might as well make the notes' likes scrollable too? Should I make changes to that as well?

**Comment by jywarren** [comment]
`style= overflow:auto` sounds great, though be sure to set it big enough to be useful -- maybe `max-height` instead of height?

**Comment by Pk13055** [comment]
@jywarren Updated as requested. Can you check them out and merge if everything looks okay?

**Comment by jywarren** [comment]
Hi -- this looks perfect; merging now. If you're looking for a new challenge, please take a look at our `help-wanted` list: https://github.com/publiclab/plots2/labels/help-wanted

Thanks!

**Comment by ebarry** [comment]
Thanks so much @Pk13055!!!!!!!!!!!!! This is a great function to have in the Question and Answer part of PublicLab.org
Really appreciating your efforts on this!

**Comment by Pk13055** [comment]
@ebarry No problem!

**Comment by jywarren** [comment]
Just to show how this is working now:

https://publiclab.org/questions/warren/12-05-2016/how-do-i-make-a-video-from-a-series-of-photos

screenshot 2016-12-26 at 11 22 48 pm

**Comment by Pk13055** [comment]
@jywarren Is this fine or do you want me to change this?

**Comment by jywarren** [comment]
Looks great, just wanted to show you that it's live on the site now!

**Comment by Pk13055** [comment]
Thanks. Feels good seeing our commits live. On Dec 27, 2016 10:06 PM, Jeffrey Warren <notifications@github.com> wrote:Looks great, just wanted to show you that it's live on the site now!

**Comment by ryzokuken** [comment]
@Pk13055 did you just see the email? 😝
